src/colab/server-picker.ts

```diff
@@ -178,10 +249,35 @@ function isAcceleratorDefined(
   return state.accelerator !== undefined;
 }
 
+function isShapeDefined(
+  state: Partial<Server>,
+): state is PartialServerWith<'shape'> {
+  return state.shape !== undefined;
+}
+
 interface VariantPick extends QuickPickItem {
   value: Variant;
 }
 
 interface AcceleratorPick extends QuickPickItem {
   value: string;
 }
+
+interface ShapePick extends QuickPickItem {
+  value: Shape;
+}
+
+function getShapePicksFromAccelerator(
+  shapesByAccelerators: Map<string, Set<Shape>>,
+  accelerator: string,
+): ShapePick[] {
+  const shapes = shapesByAccelerators.get(accelerator) ?? new Set();
+  const items: ShapePick[] = [];
```

Comment:
Nit: no need to new up the empty `Set`.

```suggestion
  const items: ShapePick[] = [];
  const shapes = shapesByAccelerators.get(accelerator);
  if (!shapes) {
    return items;
  }
```